packages/pigeon/test/dart_generator_test.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:test/test.dart';
+import 'package:pigeon/dart_generator.dart';
+import 'package:pigeon/ast.dart';
+
+void main() {
+  test('gen one class', () {
+    final Class klass = Class()
+      ..name = 'Foobar'
+      ..fields = <Field>[
+        Field()
+          ..name = 'field1'
+          ..dataType = 'dataType1'
+      ];
+    final Root root = Root()
+      ..apis = <Api>[]
+      ..classes = <Class>[klass];
+    final StringBuffer sink = StringBuffer();
+    generateDart(root, sink);
+    final String code = sink.toString();
+    expect(code, contains('class Foobar'));
```

Comment:
What about structuring the tests such that for each one there's an input file and then the expected output files.  This way, we can have end-to-end testing and less boilerplate in the test.

Reply:
I have end to end tests too, they aren't getting checked in with this PR though because it would explode the pr to 95 files and there are still some bugs to chase in e2e package.  The way I usually work is that when I get an error in the e2e tests I try to write a unit test that reproduces the error.